PetGame/Services/PetService.cs

```diff
@@ -298,31 +325,20 @@ private int CalculateHoursSinceLastActivity(IEnumerable<Activity> PastActivities
             return HoursSinceLastActivity;
         }
 
-        private int CalculateMinutesToNextAction(IEnumerable<Activity> PastActivities, int MinutesToNextAction, int HoursToCheck)
+        private DateTime TimeOfNextAction(IEnumerable<Activity> PastActivities, int MinutesToNextAction, int HoursToCheck)
         {
-            List<Activity> Activities = (List<Activity>)PastActivities;
+            List<Activity> Activities = new List<Activity>(PastActivities);
 
-            if (Activities.Count == 0)
+            var LastActivity = Activities.FirstOrDefault(x => x.Type == ActivityType.Feeding || x.Type == ActivityType.Race || x.Type == ActivityType.Training);
+
+            if (Activities.Count == 0 || LastActivity == null)
```

Comment:
```suggestion
            if (LastActivity == null)
```

The check for size is redundant, because `FirstOrDefault` will return null for empty lists.